### ISSUE TITLE
Inquiry on prepare_targets function in plan.py

### ISSUE THREAD

**Issue by tabz23** [gaoyuezhou/dino_wm#13]
1) for the below portions of the code, when goal_source is dset in the plan.yaml, why are we collecting observations from the sample_traj_segment_from_dset function and then throwing them away and collecting those same observations using self.env.rollout? can we i am confused about this implementation decision. Why the observations in the dataset used for training would be different than the ones collected from self.env.rollout. Is there anything I misunderstand or can i just directly use the observations retrieved from sample_traj_segment_from_dset? 

        else:
            # update env config from val trajs
            observations, states, actions, env_info = (
                self.sample_traj_segment_from_dset(traj_len=self.frameskip * self.goal_H + 1)
            )
            self.env.update_env(env_info)

            # get states from val trajs
            init_state = [x[0] for x in states]
            init_state = np.array(init_state)
            actions = torch.stack(actions)
            if self.goal_source == "random_action":
                actions = torch.randn_like(actions)
            wm_actions = rearrange(actions, "b (t f) d -> b t (f d)", f=self.frameskip)
            exec_actions = self.data_preprocessor.denormalize_actions(actions)
            # replay actions in env to get gt obses
            rollout_obses, rollout_states = self.env.rollout(
                self.eval_seed, init_state, exec_actions.numpy()
            )
            self.obs_0 = {
                key: np.expand_dims(arr[:, 0], axis=1)
                for key, arr in rollout_obses.items()
            }
            self.obs_g = {
                key: np.expand_dims(arr[:, -1], axis=1)
                for key, arr in rollout_obses.items()
            }
            self.state_0 = init_state  # (b, d)
            self.state_g = rollout_states[:, -1]  # (b, d)
            self.gt_actions = wm_actions 

2) Is there any example or template where goal_source=file was used in the evaluation? I'd like to use that but would prefer to have an example.

**Comment by gaoyuezhou** [comment]
Hi,

For your questions, 

1. The current code replays the action trajectories to reconstruct the goal states, but it's perfectly valid to use the observations from the dataset directly. This is more of an implementation detail than a fundamental design choice.
2. Each time you run `plan.py`, the targets will be saved to a file named `plan_targets.pkl`. If you wish you reuse the same plan targets for a different run, you can set `goal_file_path` in your plan.yaml config, and set it to the `plan_targets.pkl` file.
